src/components/Evolution.ts

```diff
@@ -0,0 +1,98 @@
+import { Component } from "eightbittr/lib/Component";
+
+import { FullScreenPokemon } from "../FullScreenPokemon";
+import { IPokemon } from "./Battles";
+import { 
+    IPokemonEvolution, IPokemonEvolutionByLevel, IPokemonEvolutionByStats, 
+    IPokemonEvolutionRequirements
+} from "./constants/Pokemon";
+
+/**
+ * Interface for evolution requirement handler.
+ */
+export interface IRequirementHandlers {
+    [method: string]: { (pokemon?: IPokemon, requirements?: IPokemonEvolutionRequirements): boolean }; 
```

Comment:
How about splitting the function declaration into its own `IRequirementHandler`? That would simplify this line.